src/main/java/com/aws/greengrass/shadowmanager/model/configuration/ShadowSyncConfiguration.java

```diff
@@ -121,69 +135,117 @@ private static void processCoreThingConfiguration(Map<String, Object> configTopi
-                    if (Utils.isEmpty(tn)) {
-                        throw new InvalidConfigurationException(String.format(UNEXPECTED_VALUE_FORMAT,
-                                CONFIGURATION_THING_NAME_TOPIC, tn));
-                    }
-                    thingName.set(tn);
-                } else {
-                    throw new InvalidConfigurationException(String.format(UNEXPECTED_TYPE_FORMAT,
-                            CONFIGURATION_THING_NAME_TOPIC, name == null ? null : name.getClass().getTypeName()));
-                }
-            }
-            Validator.validateThingName(thingName.get());
-
-            ThingShadowSyncConfiguration syncConfiguration;
-            boolean syncClassicTopic = true;
-            for (Map.Entry<String, Object> configObjectEntry : thingConfig.entrySet()) {
-                switch (configObjectEntry.getKey()) {
-                    case CONFIGURATION_CLASSIC_SHADOW_TOPIC:
-                        syncClassicTopic = Coerce.toBoolean(configObjectEntry.getValue());
-                        break;
-                    case CONFIGURATION_NAMED_SHADOWS_TOPIC:
-                        if (configObjectEntry.getValue() instanceof List) {
-                            List<String> namedShadows = Coerce.toStringList(configObjectEntry.getValue());
-                            for (String namedShadow : namedShadows) {
-                                Validator.validateShadowName(namedShadow);
-                                syncConfiguration = ThingShadowSyncConfiguration.builder()
-                                        .thingName(thingName.get())
-                                        .shadowName(namedShadow)
-                                        .build();
-                                syncConfigurationSet.add(syncConfiguration);
-                            }
-                        } else {
-                            throw new InvalidConfigurationException(String.format(UNEXPECTED_TYPE_FORMAT,
-                                    configObjectEntry.getKey(), configObjectEntry.getClass().getTypeName()));
-                        }
-                        break;
-                    default:
-                        // Do nothing here since we want to be lenient with unknown fields.
-                        break;
-                }
-            }
-            if (syncClassicTopic) {
-                syncConfiguration = ThingShadowSyncConfiguration.builder()
-                        .thingName(thingName.get())
-                        .shadowName(CLASSIC_SHADOW_IDENTIFIER)
-                        .build();
-                syncConfigurationSet.add(syncConfiguration);
-            }
+            processThingShadowSyncConfiguration(syncConfigurationSet, thingConfig, thingName);
+        } else {
+            throw new InvalidConfigurationException(String.format(UNEXPECTED_TYPE_FORMAT,
+                    CONFIGURATION_SHADOW_DOCUMENTS_TOPIC, thingConfigObject.getClass().getTypeName()));
```

Comment:
does this need to use `CONFIGURATION_SHADOW_DOCUMENTS_MAP_TOPIC` for the error message here?